python/foglamp/common/storage_client/storage_client.py

```diff
@@ -635,20 +638,27 @@ def disconnect(self):
             raise TypeError("Provided data to insert must be a valid JSON")
 
         post_url = '/storage/table/{tbl_name}'.format(tbl_name=tbl_name)
-
-        try:
-            url = 'http://' + self.base_url + post_url
-            async with aiohttp.ClientSession() as session:
-                async with session.post(url, data=data) as resp:
+        attempt_count = 1
+        while True:
```

Comment:
No, attempts thing are meant for caller not the storage client code.

Reply:
Nope. Here caller IS the async storage client method, hence this exception should be handled here only and NOT by the caller of the async storage client method and should inform the caller of the failure only when a timeout occurs.